### ISSUE TITLE
Sentiment analysis model

### ISSUE THREAD

**Issue by rlucatoor** [tanaos/artifex#20]
## What is a Sentiment Analysis model

A Sentiment Analysis model is a text classification model that predicts a text's emotional tone by assigning a label to it. Classic labels are `positive`, `neutral` and `negative`, but they can be extended to include `very-positive` and `very-negative`.

## What needs to be done

Implement a Sentiment Analysis model that exposes the following public methods:

### `.train()` method:

```python
.train(extended: Optional[bool] = False, output_path: Optional[str] = None, num_samples: int = config.DEFAULT_SYNTHEX_DATAPOINT_NUM, num_epochs: int = 3)`
```

where the `extended` argument determines whether 3 (`extended=False`) or 5 (`extended=True`) text sentiment classes should be used.

The method should then train the model to assign one of the 3 (or 5) labels to the provided text, based on its sentimental tone. The `BaseModel._generate_synthetic_data()` method should be used to generate synthetic training data.

### `.__call__(text: Union[str, list[str])`

- If a `str` is provided, the model should return:
    - Any one of the 3 possible strings `"positive"`, `"neutral"` or `"negative"`, if training was performed with `extended=False`,
    - Any one of the 5 possible strings `"very-positive"`, `"positive"`, `"neutral"`, `"negative"` and `"very-negative"` if training was performed with `extended=True`,

- If a `list[str]` is provided, the model should return a list of strings, one per sentence.

### `.load(model_path: str)`

Loads a pre-trained Sentiment Analysis model from the specified path.
